target/mips/helper.c

```diff
@@ -80,6 +80,26 @@ int r4k_map_address(CPUMIPSState *env, hwaddr *physical, int *prot,
     uint16_t ASID = env->CP0_EntryHi & env->CP0_EntryHi_ASID_mask;
     int i;
 
+#if defined(TARGET_CHERI)
+    unsigned int gclg;
+    if (address < 0x4000000000000000) {
+        /* useg, xuseg */
+        gclg = !!(env->CP0_EntryHi & (1UL << CP0EnHi_CLGU));
```

Comment:
It took me a while to see what was different between the ifs. Maybe easier to read if each case assigns to somthing like `bit_to_test` and `bool gclg = !!(env->CP0_EntryHi & (1UL << bit_to_test))` after the if/else statements?